scripts/dockerlib.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+# Author Gabriel Hartmann <gabhart@microsoft.com>
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Open Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#--------------------------------------------------------------------------
+
+file_name=${0##*/}
+
+timestamp() {
+    date +"%Y-%m-%d %H:%M:%S"
+}
+
+log() {
+    echo "$file_name $(timestamp): $1"
+}
+
+json_val() {
+    python -c 'import json,sys;obj=json.load(sys.stdin);print obj'$1'';
+}
+
+json_dump() {
+    python -c 'import json,sys;obj=json.load(sys.stdin);print json.dumps(obj'$1')';
+}
+
+yaml_dump() {
+    python -c 'import json,yaml,sys;data=json.load(sys.stdin);print yaml.safe_dump(data, default_flow_style=False)'
+}
+
+get_script_dir() {
+    echo $(cd $(dirname $0); pwd)
+}
+
+get_log_dir() {
+    echo $(cat $(get_script_dir)/../HandlerEnvironment.json | \
+	json_val '[0]["handlerEnvironment"]["logFolder"]')
+}
+
+get_config_dir() {
+    echo $(cat $(get_script_dir)/../HandlerEnvironment.json | \
+	json_val '[0]["handlerEnvironment"]["configFolder"]')
+}
+
+get_status_dir() {
+    echo $(cat $(get_script_dir)/../HandlerEnvironment.json | \
+	json_val '[0]["handlerEnvironment"]["statusFolder"]')
+}
+
+get_log_file() {
+    echo $(get_log_dir)/docker-handler.log
```

Comment:
might be do these returns safer like `echo "$(get_log_dir)/docker-handler.log"` with quotes.